tests/components/broadlink/test_device.py

```diff
@@ -0,0 +1,107 @@
+"""Tests for Broadlink devices."""
+import broadlink.exceptions as blke
+import pytest
+
+from homeassistant.components.broadlink.const import DOMAIN
+from homeassistant.components.broadlink.device import BroadlinkDevice
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.setup import async_setup_component
+
+from . import pick_device
+
+from tests.async_mock import AsyncMock, MagicMock, patch
+
+
+async def test_device_setup(hass):
+    """Test a successful setup."""
+    device = pick_device(1)
+    mock_api = device.get_mock_api()
+    mock_entry = device.get_mock_entry()
+    device_entry = BroadlinkDevice(hass, mock_entry)
+    await async_setup_component(hass, DOMAIN, {})
+
+    with patch("broadlink.gendevice", return_value=mock_api), patch.object(
+        hass.config_entries, "async_forward_entry_setup"
+    ) as mock_forward:
+        assert await device_entry.async_setup() is True
```

Comment:
We don't want to call this directly. We just want to set up the integration.

Reply:
How about this:
```python3
async def test_device_setup(hass):
    """Test a successful setup."""
    device = pick_device(1)
    mock_api = device.get_mock_api()
    mock_entry = device.get_mock_entry()
    mock_entry.add_to_hass(hass)

    with patch("broadlink.gendevice", return_value=mock_api), patch.object(
        hass.config_entries, "async_forward_entry_setup"
    ) as mock_forward:
        assert await hass.config_entries.async_setup(mock_entry.entry_id)

    device_entry = hass.data[DOMAIN].devices[mock_entry.entry_id]
    assert device_entry.api is mock_api
    assert device_entry.authorized is True
    assert mock_api.auth.call_count == 1
    assert mock_api.get_fwversion.call_count == 1
    assert len(mock_forward.mock_calls) == 3
    forward_entries = {c[1][1] for c in mock_forward.mock_calls}
    assert forward_entries == {"remote", "sensor", "switch"}
```

Reply:
We don't want to assert anything about the `device_entry` directly. We just want to assert things in the home assistant core state. That's either the Entity state machine, the entity registry, the device registry or the config entry.

Reply:
Ok.

Reply:
Done.